core/src/main/java/brooklyn/location/basic/SshMachineLocation.java

```diff
@@ -876,7 +877,8 @@ public boolean isSshable() {
         String cmd = "date";
         try {
             try {
-                Socket s = new Socket(getAddress(), getPort());
+                Socket s = new Socket();
+                s.connect(new InetSocketAddress(getAddress(), getPort()), SSHABLE_CONNECT_TIMEOUT);
```

Comment:
Makes sense. Could overload `isSshable()` to allow a custom timeout to be passed in, but not sure where we'd make use of that. So happy to stick with your change.